core/src/solution_submission/retry.rs

```diff
@@ -6,9 +6,51 @@ use ethcontract::{
     errors::{ExecutionError, MethodError},
     U256,
 };
+use futures::future::{BoxFuture, FutureExt as _};
 
 use super::MIN_GAS_PRICE_INCREASE_FACTOR;
 
+pub struct Args<'a> {
+    pub contract: &'a (dyn StableXContract + Sync),
+    pub batch_index: u32,
+    pub solution: Solution,
+    pub claimed_objective_value: U256,
+    pub gas_price_estimating: &'a (dyn GasPriceEstimating + Sync),
+    pub gas_price_cap: U256,
+    pub nonce: U256,
+}
+
+pub trait RetryingWithGasPriceIncrease {
```

Comment:
I would argue that the trait itself is really just capable of submitting a transaction (it's only invoked once from the caller) and therefore suggest to name it `TransactionSending` or `SolutionTransactionSending` if you want to keep the logic tx specific.

We then have a concrete implementation that does so by retrying with gas increase.

Reply:
For now the intention is specific to the solution as is seen in the parameters of the one function the trait has. So I don't think it makes sense to give it a more generic name.

Reply:
I don't think the fact that it is retrying is specific to the arguments. Specifically because the gas_price_estimating does not need to part of Args even if we keep it tx specific.

Reply:
Implemented rename and moved the two args into the struct. Has nice side effect that solution submission no longer has to store the gas price estimating.